test/runtests.jl

```diff
@@ -44,61 +45,70 @@ setprogress!(false)
 
 include(pkgdir(Turing)*"/test/test_utils/AllUtils.jl")
 
+# Collect timing and allocations information to show in a clear way.
+const TIMEROUTPUT = TimerOutputs.TimerOutput()
+macro timeit_include(path::AbstractString)
+    return :(@timeit TIMEROUTPUT $path include($path))
+end
+
 @testset "Turing" begin
     @testset "essential" begin
-        include("essential/ad.jl")
+        @timeit_include("essential/ad.jl")
     end
 
     @testset "samplers (without AD)" begin
-        include("inference/AdvancedSMC.jl")
-        include("inference/emcee.jl")
-        include("inference/ess.jl")
-        include("inference/is.jl")
+        @timeit_include("inference/AdvancedSMC.jl")
+        @timeit_include("inference/emcee.jl")
+        @timeit_include("inference/ess.jl")
+        @timeit_include("inference/is.jl")
     end
 
     Turing.setrdcache(false)
     for adbackend in (:forwarddiff, :tracker, :reversediff)
-        Turing.setadbackend(adbackend)
-        @info "Testing $(adbackend)"
-        start = time()
-        @testset "inference: $adbackend" begin
-            @testset "samplers" begin
-                include("inference/gibbs.jl")
-                include("inference/gibbs_conditional.jl")
-                include("inference/hmc.jl")
-                include("inference/Inference.jl")
-                include("contrib/inference/dynamichmc.jl")
-                include("contrib/inference/sghmc.jl")
-                include("inference/mh.jl")
+        @timeit TIMEROUTPUT "inference: $adbackend" begin
+            Turing.setadbackend(adbackend)
+            @info "Testing $(adbackend)"
+            start = time()
```

Comment:
Maybe not needed anymore?

```suggestion
```

Reply:
There is still the 
```julia
# Useful for figuring out why CI is timing out.
@info "Tests for $(adbackend) took $(time() - start) seconds"
```
comment, so I guess that it is still useful